mesonbuild/compilers/asm.py

```diff
@@ -47,6 +47,14 @@ def get_werror_args(self) -> T.List[str]:
     def get_output_args(self, outputname: str) -> T.List[str]:
         return ['-o', outputname]
 
+    def unix_args_to_native(self, args: T.List[str]) -> T.List[str]:
+        outargs = []
+        for arg in args:
+            if arg == '-pthread':
```

Comment:
We'll probably have to filter more than that, wondering if we should start from a white-list instead of black-list. OTOH, this is already a good start and we can expand when people report issues.

Reply:
TBH the real issue is that nasm is compiler-adjacent, but not actually a C/C++ compiler and cannot accept standard dependency args. If only we handled dependency compile args per language...

Reply:
> If only we handled dependency compile args per language...

That's unfortunate indeed. In the meantime I think this PR is fine, that's also was msvc compiler does: https://github.com/mesonbuild/meson/blob/master/mesonbuild/compilers/mixins/visualstudio.py#L231.